src/plugins/discourse/fetch.js

```diff
@@ -0,0 +1,212 @@
+      const {apiUsername, apiKey} = this.options;
+      throw new Error(
+        `got status 403; bad api username (${apiUsername}) or key (${apiKey})`
+      );
+    }
+    if (response.status !== 200) {
+      throw new Error(`not OK status ${response.status} on ${response.url}`);
+    }
+    const json = await response.json();
+    if (json.topic_list.topics.length === 0) {
+      throw new Error(`no topics! got ${stringify(json)} as latest topics.`);
+    }
+    return json.topic_list.topics[0].id;
+  }
+
+  async latestPosts(): Promise<$ReadOnlyArray<Post>> {
+    const response = await this._fetch("/posts.json");
+    if (response.status === 404) {
+      throw new Error(`404 from ${response.url}; maybe bad serverUrl?`);
+    }
+    if (response.status === 403) {
+      const {apiUsername, apiKey} = this.options;
+      throw new Error(
+        `got status 403; bad api username (${apiUsername}) or key (${apiKey})`
+      );
+    }
+    if (response.status !== 200) {
+      throw new Error(`not OK status ${response.status} on ${response.url}`);
+    }
+    const json = await response.json();
+    return json.latest_posts.map(parsePost);
+  }
+
+  async topicWithPosts(id: TopicId): Promise<TopicWithPosts | null> {
+    const response = await this._fetch(`/t/${id}.json`);
+    if (response.status === 404) {
+      // Not sure why this happens, but a topic can sometimes 404.
+      // We should just consider it unreachable.
+      // Here is an example: https://discourse.sourcecred.io/t/116
+      return null;
+    }
+    if (response.status === 403) {
+      // Probably this topic is hidden or deleted.
+      // Just consider it unreachable.
+      // If the issue is that the user provided bad keys, then
+      // they will get a more helpful error when they try to get the latest
+      // topic id.
+      return null;
+    }
+    if (response.status !== 200) {
+      throw new Error(`not OK status ${response.status} on ${response.url}`);
+    }
+    const json = await response.json();
+    const posts = json.post_stream.posts.map(parsePost);
+    const topic: Topic = {
+      id: json.id,
+      title: json.title,
+      timestampMs: +new Date(json.created_at),
+      authorUsername: json.details.created_by.username,
+    };
+    return {topic, posts};
+  }
+
+  async post(id: PostId): Promise<Post | null> {
+    const response = await this._fetch(`/posts/${id}.json`);
+    if (response.status === 404) {
+      // Since topics can 404, I assume posts can too.
+      return null;
+    }
+    if (response.status === 403) {
+      // Probably this post is hidden or deleted.
+      return null;
+    }
+    if (response.status !== 200) {
+      throw new Error(`not OK status ${response.status} on ${response.url}`);
+    }
+    const json = await response.json();
+    return parsePost(json);
+  }
+}
+
+function parsePost(json: any): Post {
+  return {
+    id: json.id,
+    timestampMs: +new Date(json.created_at),
```

Comment:
This is [the same as][1] `Date.parse(json.created_at)`.

[1]: https://tc39.es/ecma262/#sec-date-value